src/vis/vis.js

```diff
@@ -137,17 +136,21 @@ var VisModel = Backbone.Model.extend({
     var windshaftClient = new WindshaftClient(windshaftSettings);
 
     // Create the public Analysis Factory
-    this.analysis = new AnalysisService({
+    // TODO: use only AnalysisService static methods without instance
+    this.analysisService = new AnalysisService({
       apiKey: this.get('apiKey'),
       authToken: this.get('authToken'),
       analysisCollection: this._analysisCollection,
       vis: this
     });
 
-    var layersFactory = new LayersFactory({
-      visModel: this,
-      windshaftSettings: windshaftSettings
-    });
+    // Public namespace exposing public methods.
+    this.analysis = {
```

Comment:
Should we add a couple of tests for this namespace?

Reply:
IMO not yet, this functionality is covered by another tests.

At the end of the project we can revisit the public api and write exclusive tests for it